backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/config/SchemaService.java

```diff
@@ -150,21 +152,22 @@ SchemaAPI addSchema(@NonNull String schemaId, @Nullable String label,
         return Optional.empty();
     }
 
-    public List<SchemaAPI> listSchemas() {
-        List<SchemaAPI> result = new ArrayList<>();
-        schemaRepo.findAll().forEach(dbS -> result.add(SchemaAPI.from(dbS)));
-        return result;
+    String getDid() {
+        return id.getMyDid() == null ? "" : id.getMyDid();
     }
 
-    public List<SchemaAPI> listSchemas(@NonNull String did) {
-        String creatorDid = did;
-        if (did.startsWith(rc.getLedgerPrefix())) {
-            creatorDid = AriesStringUtil.getLastSegment(did);
+    String getCreatorDid() {
```

Comment:
I think we need to be careful here, because aca-py now also starts to require fully qualified dids from version 0.7.0 onwards, because they now also support did:key. So they start to expect (and return) something like `did:sov:123`, `did:key:123`. I started work to get rid of the universal resolver and instead using the resolver capabilities of aca-py. Then we are more in sync with what there are doing. Long story short, we need to get rid of the ledger prefix and keep prefixing logic in one place. So I would at least move those methods to the Identity service.

Reply:
Does this make sense? Basically what you want is the did with or without the prefix. So this should definitely the responsibility of the Identity service. Then its more like getMyDid() getMyDidLastSegment().

Reply:
I will move all of this into Identity, to keep all the logic in one place.